### PR TITLE
UI-420 Update package to match ui-components-predix

### PR DESCRIPTION
**Pull request by mallik-svmx**
- Configured jest/enzyme setup and turn tests to green status
- Pulling tests to respective component directory
- Updated ES linting rules and linting cleanp from auto fix only, individual per component stories should take care of remaining linting errors
- Updated all required package dependencies
- Please, make sure to "npm install" after you sync this PR

### CODE REVIEW

File: lib/components/Datepicker/DateInput.js [hunk 1]
```diff
@@ -244,9 +244,9 @@ export default class DateInput extends Component {
     const pprops = props;
     delete pprops.onValueChange;
     return (
-      <div className="slds-input-has-icon slds-input-has-icon--right">
+      <div className='slds-input-has-icon slds-input-has-icon--right'>
```
Comment (joshsylvester) [inline lib/components/Datepicker/DateInput.js]: this is actually an incorrect change - we should use double quotes for html / JSX properties... multiple times in this file.

File: lib/components/Button/Button.test.js [hunk 2]
```diff
@@ -76,21 +76,25 @@ describe('Button', () => {
...
   it('should render button with icon more', () => {
     const iconMore = 'down';
-    const wrapper = shallow(<Button iconMore={ iconMore } />);
-    expect(wrapper.contains(
-      <ButtonIcon icon={ iconMore } align='right' size='small' />
-    )).toBe(true);
+    const wrapper = shallow(<Button iconMore={iconMore} />);
+    expect(
+      wrapper.contains(
+        <ButtonIcon icon={iconMore} align='right' size='small' />,
```
Comment (joshsylvester) [inline lib/components/Button/Button.test.js]: double quotes for JSX properties

**Comment by mallik-svmx** [comment]
I missed a commit, please hold-on your review comments. thanks!

**Comment by mallik-svmx** [comment]
@joshsylvester - Review comments addressed. Please review. thanks!

fyi - @sheela-svmx  @guohonghanusa

**Comment by mallik-svmx** [comment]
Two approval, merging.

**Pull request merged**